tests/checktags_tests.py

```diff
@@ -0,0 +1,495 @@
+            match_querystring = True,
+            responses = [
+                httpretty.Response( body = """
+                    <collection matches="1">
+                      <request id="254684">
+                        <action type="submit">
+                          <source project="Base:System" package="timezone" rev="481ecbe0dfc63ece3a1f1b5598f7d96c"/>
+                          <target project="openSUSE:Factory" package="timezone"/>
+                        </action>
+                        <state name="review" who="maintbot" when="2014-10-08T11:55:56">
+                          <comment>...</comment>
+                        </state>
+                        <description> ... </description>
+                      </request>
+                    </collection>
+                    """),
+                httpretty.Response( body = """
+                    <collection matches="1">
+                      <request id="254684">
+                        <action type="submit">
+                          <source project="Base:System" package="timezone" rev="481ecbe0dfc63ece3a1f1b5598f7d96c"/>
+                          <target project="openSUSE:Factory" package="timezone"/>
+                        </action>
+                        <state name="new" who="maintbot" when="2014-10-08T11:55:56">
+                          <comment>...</comment>
+                        </state>
+                        <description> ... </description>
+                      </request>
+                    </collection>
+                    """)
+                ])
+
+        result = { 'state_accepted' : None }
+
+        def change_request(result, method, uri, headers):
+            u = urlparse.urlparse(uri)
+            if u.query == 'newstate=accepted&cmd=changereviewstate&by_user=maintbot':
+                result['state_accepted'] = True
+            elif u.query == 'newstate=declined&cmd=changereviewstate&by_user=maintbot':
+                result['state_accepted'] = False
+            return (200, headers, '<status code="ok"/>')
+
+        httpretty.register_uri(httpretty.POST,
+            APIURL + "/request/770001",
+            body = lambda method, uri, headers: change_request(result, method, uri, headers))
+
+        # first time request is in in review
+        self.checker.set_request_ids(['770001'])
+        self.checker.check_requests()
+
+        self.assertFalse(result['state_accepted'])
+
+        # second time request is in state new so we can accept
+        # first time request is in in review
+        self.checker.set_request_ids(['770001'])
+        self.checker.check_requests()
+
+        self.assertFalse(result['state_accepted'])
+
+    def test_non_maintainer_submit(self):
```

Comment:
hehe, never tried to hide it. I copied and adapted some tests from maintenance_tests and factory_source_tests :)